src/components/screens/editor/editor-table.tsx

```diff
@@ -0,0 +1,369 @@
+ */
+const EditorScreenTable: FC<Props> = ({
+  loading,
+  metaforms,
+  metaformVersions,
+  deleteMetaformOrVersion,
+  setSelectedId,
+  editMetaform
+}) => {
+  // TODO: Currently API doesn't return metadata (created/modified dates etc) for Metaforms. 
+  // That needs to be changed and after that, this components version row functionality need slight refactoring.
+  const [ popoverAnchorElement, setPopoverAnchorElement ] = useState<HTMLButtonElement | null>(null);
+  const [ deleteDialogOpen, setDeleteDialogOpen ] = useState<boolean>(false);
+  const [ popoverOpen, setPopoverOpen ] = useState<boolean>(false);
+
+  /**
+   * Handles popover menu opening
+   */
+  const handleMenuOpen = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setPopoverAnchorElement(event.currentTarget);
+    setPopoverOpen(!popoverOpen);
+    setSelectedId(event.currentTarget.id);
+  };
+  
+  /**
+   * Handles popover menu close
+   */
+  const handleMenuClose = () => {
+    setPopoverAnchorElement(null);
+    setPopoverOpen(!popoverOpen);
+    setSelectedId(undefined);
+  };
+
+  /**
+   * Handles delete confirmation
+   */
+  const handleDelete = () => {
+    deleteMetaformOrVersion();
+    setSelectedId(undefined);
+    setDeleteDialogOpen(false);
+    handleMenuClose();
+  };
+
+  /**
+   * Renders delete confirm dialog
+   */
+  const renderDeleteConfirm = () => (
+    <ConfirmDialog
+      onClose={ () => setSelectedId(undefined) }
+      onCancel={ () => setDeleteDialogOpen(!deleteDialogOpen) }
+      onConfirm={ handleDelete }
+      cancelButtonText={ strings.generic.cancel }
+      positiveButtonText={ strings.generic.confirm }
+      title={ strings.editorScreen.confirmDeleteVersionTitle }
+      text={ strings.editorScreen.confirmDeleteVersionText }
+      open={ deleteDialogOpen }
+    />
+  );
+
+  /**
+   * Handles edit button click
+   */
+  const handleEditClick = () => {
+    handleMenuClose();
+    editMetaform();
+  };
+  
+  /**
+   * Renders version menu
+   */
+  const renderVersionMenu = () => (
+    <Popover
+      open={ popoverOpen }
+      anchorEl={ popoverAnchorElement }
+      onClose={ handleMenuClose }
+      anchorOrigin={{
+        vertical: "bottom",
+        horizontal: "right"
+      }}
+      transformOrigin={{
+        vertical: "top",
+        horizontal: "right"
+      }}
+    >
+      <List sx={{ p: theme.spacing(0.5) }}>
+        <ListItem sx={{ p: theme.spacing(0.5) }}>
+          <ListItemButton
+            sx={{ p: 0 }}
+            onClick={ () => setDeleteDialogOpen(!deleteDialogOpen) }
+          >
+            <ListItemIcon sx={{ minWidth: 0, mr: theme.spacing(1) }}>
+              <Delete/>
+            </ListItemIcon>
+            <ListItemText primary={ strings.generic.delete }/>
+          </ListItemButton>
+        </ListItem>
+        <ListItem sx={{ p: theme.spacing(0.5) }}>
+          <ListItemButton
+            sx={{ p: 0 }}
+            onClick={ handleEditClick }
+          >
```

Comment:
Could there be just link to editor page?

Reply:
Done.